src/xml/ConfigParser.cpp

```diff
@@ -214,7 +214,10 @@ void ConfigParser::connectTags(const ConfigurationContext &context, std::vector<
 
   for (auto &subtag : SubTags) {
     std::string expectedName = (subtag->m_Prefix.length() ? subtag->m_Prefix + ":" : "") + subtag->m_Name;
-    const auto  tagPosition  = std::find_if(
+    PRECICE_CHECK(expectedName != "solver-interface",
+                  "This configuration contains the tag <solver-interface>, meaning it was written for a preCICE version 1 or 2. "
+                  "Please check that you are using the correct case or preCICE version.");
```

Comment:
```suggestion
                  "Please check that you are using the intended configuration file and preCICE version.");
```

Reply:
Let's tailor this part explicitly to a new user, who mistakenly uses the incorrect version of the tutorials / quickstart.

* Please check that the expected version of your simulation case matches this version of preCICE.

* Please ensure that the used simulation case was made for this version of preCICE.

* Please ensure that the used simulation case is compatible with this version of preCICE.

* Are you using the correct version of your simulation case? Has this simulation case been updated to this version of preCICE?

Reply:
```suggestion
                  "Are you using the correct version of your simulation case? Has this simulation case been updated to this version of preCICE?");
```